Public/Sdk/Public/Managed/Tools/AppHostPatcher/AppHostPatcher.dsc

```diff
@@ -3,33 +3,31 @@
 
 import {Artifact, Cmd, Transformer} from "Sdk.Transformers";
 import * as Managed from "Sdk.Managed";
+import * as Shared from "Sdk.Managed.Shared";
 import * as MacOS from "Sdk.MacOS";
+import * as Frameworks from "Sdk.Managed.Frameworks";
 
 export declare const qualifier: Managed.TargetFrameworks.CurrentMachineQualifier;
 
 const pkgContents = importFrom("BuildXL.Tools.AppHostPatcher").Contents.all;
 
+const patcherExecutable = Context.getCurrentHost().os === "macOS"
+    ? Managed.nativeExecutable({ 
+        assemblyName: "NativeAppHostPatcher",
+        sources: globR(d`.`, "*.cs")
+      })
+    : pkgContents.getFile(r`tools/win-x64/AppHostPatcher.exe`);
+
 const patcher: Transformer.ToolDefinition = {
-    exe: pkgContents.getFile(r`tools/${qualifier.targetRuntime}/${"AppHostPatcher" + (qualifier.targetRuntime === "win-x64" ? ".exe" : "")}`),
-    dependsOnWindowsDirectories: true,
-    untrackedDirectoryScopes: [
-        ...addIfLazy(Context.getCurrentHost().os === "macOS", () => [
-            ...MacOS.untrackedSystemFolderDeps,
-            d`${Context.getMount("UserProfile").path}/Microsoft/BuildXL/RestrictedTemp`,
-        ])
-    ],
-    untrackedFiles: [
-        ...addIfLazy(Context.getCurrentHost().os === "macOS", () => MacOS.untrackedFiles)
-    ],
+    exe: patcherExecutable,
+    dependsOnCurrentHostOSDirectories: true,

```

Comment:
I assume this means things like `untrackedSystemFolderDeps` are now automatically added in our backend code, we have a lot of other `untrackedDirectoryScopes` in other tools, do you mind making a pass over that and removing explicit calls and replacing them with `dependsOnCurrentHostOSDirectories` if that is the cleaner way?